### PR TITLE
add images

### PR DESCRIPTION
**Pull request by xswoodsx**
### Context 
Images needed to be converted to govspeak in order to render on the ecf railsapp.

### Changes proposed in this pull request
Adding functions that retrieve and convert the alt, src and url for images hosted on the WP site. 

### Guidance to review 
The _findCaptionText_ function is required to find captions that are nested within the figcaption. It runs recursively until it finds the value as not all values can be found at the same point in the nesting. 
The _fixImages_ function has been added as the \n gets replaced with whitespace in the initial conversion of the imageData. A new line after $Figure and before the $EndFigure are required to work on the railsapp using the ECF-govspeak as it currently is.

### CODE REVIEW

File: parsers/handleWordpressComponents.js [hunk 1]
```diff
@@ -74,6 +74,56 @@ export function fixAccordions() {
...
+      const arrayOfImageData = node.children.filter((imageData) => imageData.tagName === 'img');
+      const { src } = arrayOfImageData[0].properties;
+      const { alt } = arrayOfImageData[0].properties;
+
+      const figureCaption = node.children.filter((element) => element.tagName === 'figcaption');
+
+      let captionText = '';
+      if (figureCaption.length > 0) {
+        captionText = findCaptionText(figureCaption, 'text');
+      } else captionText = '';
```
Comment (kicferk1) [inline parsers/handleWordpressComponents.js]: That line seems excessive - `captionText` is already defined as empty string three lines up.

File: parsers/handleWordpressComponents.js [hunk 2]
```diff
@@ -74,6 +74,56 @@ export function fixAccordions() {
...
+  }
+  return figCaptionValue;
+};
+
+export function findImageData() {
+  return (tree) => {
+    visit(tree, (node) => node.type === 'element' && node.children.some((n) => n.tagName === 'img'), (node) => {
+      const arrayOfImageData = node.children.filter((imageData) => imageData.tagName === 'img');
+      const { src } = arrayOfImageData[0].properties;
+      const { alt } = arrayOfImageData[0].properties;
```
Comment (kicferk1) [inline parsers/handleWordpressComponents.js]: You can do
```
const { src, alt } = arrayOfImageData[0].properties;
```
Out of curiosity, that array will never have more than one element, right? Might be worth running it for all our in files just to be sure.

File: parsers/handleWordpressComponents.js [hunk 3]
```diff
@@ -74,6 +74,56 @@ export function fixAccordions() {
...
+
+      const figureCaption = node.children.filter((element) => element.tagName === 'figcaption');
```
Comment (kicferk1) [inline parsers/handleWordpressComponents.js]: `figureCaptionNodes`

File: parsers/handleWordpressComponents.js [hunk 4]
```diff
@@ -74,6 +74,56 @@ export function fixAccordions() {
   };
 }
 
+const findCaptionText = (figureCaption, key) => {
```
Comment (kicferk1) [inline parsers/handleWordpressComponents.js]: `key` is always `'text'` - I think we can hard-code it. 
And let's call `figureCaption` `figureCaptionNodes` - so it's obvious it's an array of nodes.

File: parsers/handleWordpressComponents.js [hunk 5]
```diff
@@ -74,6 +74,56 @@ export function fixAccordions() {
...
+  let figCaptionValue = '';
+  const nestedFigCap = figureCaption.filter((o) => o.type === key);
```
Comment (kicferk1) [inline parsers/handleWordpressComponents.js]: That name is actually lying - these are the nodes in `figureCaptionNodes` that are of type `'text'` - so they are in fact not nested, they are as `shallow` as it gets :) 

`figureCaptionTextNodes`?

**Review by kicferk1 — COMMENTED** [review]
Mostly renaming changes :)

File: parsers/handleWordpressComponents.js (hunk 1 above)
Comment (xswoodsx) [reply parsers/handleWordpressComponents.js]: Definitely excessive!

File: parsers/handleWordpressComponents.js (hunk 2 above)
Comment (xswoodsx) [reply parsers/handleWordpressComponents.js]: That's right yeah but i tried running all the files before hand to make sure it was doing as expected for TF, edt etc.

File: parsers/handleWordpressComponents.js (hunk 4 above)
Comment (xswoodsx) [reply parsers/handleWordpressComponents.js]: brill, will put that in

File: parsers/handleWordpressComponents.js (hunk 5 above)
Comment (xswoodsx) [reply parsers/handleWordpressComponents.js]: That makes sense, will add that change to :)

**Review by kicferk1 — APPROVED** [review]

**Pull request merged**